### DISCUSSION TITLE
PointerPressed

### DISCUSSION THREAD

**Discussion by Ruttie2006** [AvaloniaUI/Avalonia#9394] [Q&A]
I have a rectangle somewhere in my window, and I want to know when the mouse is clicked when on that rectangle.
However, the PointerPressed event never fires.
Is there a property that makes it so the PointerPressed event of child objects also fires, or something similar?

**Comment by Ruttie2006** [comment]
The logical tree of the rectangle in question is this:
![image](https://user-images.githubusercontent.com/80394237/200862640-2657e259-10f6-4493-b656-5d0cb07ea14b.png)

**Comment by timunie** [comment]
Does the Rectangle has a Background? Transparent is okay, but null will not work. Also make sure your control is hittest visible

**Reply by Ruttie2006** [reply]
The rectangle has a black fill (I assume you mean that, as the rectangle class does not have a background property), and I saw that the default value for hittest visible is true, while I haven't touched it anywhere.
Should explicitly I set hittest visible to true for everything in the tree?

**Reply by timunie** [reply]
Use devtools via F12 in debug mode to inspect the visual tree and see if any parent has an issue. Otherwise I would need a simple sample.
